### PR TITLE
Refactor event calendar flow with new modular components and remove r…

### PR DESCRIPTION
**Pull request by esteban737**
…epeated date conversion function

Fixes #

## Description

Refactor calendar and event related code by creating modular components (EventPanel & DefaultAgenda). Changed Events component into a functional component and removed repeated instances of the convertNumtoDate function. 

## Types of changes

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other Refactoring

## Checklist:

- [x] My code follows the code style of this project found on [Contribution guidelines](https://github.com/SHPEUCF/shpeucfapp/wiki/Contribution-guidelines).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/screens/Events/Events.js [hunk 1]
```diff
@@ -1,82 +1,63 @@
...
+const { height } = Dimensions.get('screen');
 
-class Events extends Component {
-	constructor(props) {
-		super(props);
+const getTodaysDate = () => {
+	let date = new Date();
+	let month = prepend0((date.getMonth() + 1).toString());
+	let year = date.getFullYear();
+	let day = prepend0(date.getDate().toString());
```
Comment (idelmr) [inline src/screens/Events/Events.js:15]: Maybe we should even make this a function...
```js
const getDate(format) => [day, month, year]
/**
  * param: format ('number' || 'string')
  * returns array of [day, month, year] as
  * -> numbers: [1, 1, 2021]
  * -> words:   ['January', '1', '2021']
*/
```

---

```suggestion
	let date = new Date();
	let month = (date.getMonth() + 1).toString().padStart(2, '0');
	let year = date.getFullYear();
	let day = date.getDate().toString().padStart(2, '0');
```

File: src/utils/events.js [hunk 2]
```diff
@@ -37,6 +57,20 @@ export function convertMilitaryToStandardTime(militaryTime) {
...
+ */
+
+export function convertNumToDate(date) {
+	let [, month, day] = date.split('-');
+
+	return `${shortMonths[Number(month) - 1]} ${day}`;
+}
+
 /**
  * @description Prepends a 0 to any single digit number, otherwise returns the original number.
```
Comment (idelmr) [inline src/utils/events.js:75]: Eliminate this god-forsaken function

File: src/screens/Events/Events.js [hunk 3]
```diff
@@ -1,82 +1,63 @@
...
-	renderButton() {
-		const { activeUser } = this.props;
+export const Events = () => {
+	const { mainBackgroundColor, secondaryBackgroundColor, fullFlex } = styles;
+	const [isEventFormVisible, toggleEventFormVisibility] = useState(false);
+	const [targetDate, changeTargetDate] = useState(getTodaysDate());
+	const { user: { activeUser: { color, privilege } }, events: { sortedEvents } } = useSelector(state => state);
 
+	const renderButton = () => {
 		return (
```
Comment (idelmr) [inline src/screens/Events/Events.js:27]: Remove curly braces (replace with paren) and `return` statement

File: src/utils/EventPanel.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+			<TouchableOpacity onPress = { () => navigation.push('EventDetails', eventRef) }>
+				<View style = { [itemContainer, { backgroundColor }] }>
+					<Text style = { [{ fontWeight: 'bold' }, textColor] }>{ committee ? committee + ': ' + name : type + ': ' + name }</Text>
+					<Text style = { textColor }>Time: { startTime } - { endTime }</Text>
+					<Text style = { textColor }>Location: { location }</Text>
+				</View>
+			</TouchableOpacity>
+		);
+	}
+
```
Comment (idelmr) [inline src/utils/EventPanel.js:36]: Eliminate empty line

File: src/utils/EventPanel.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+			eventTextStyle,
+			leaderboardArrow,
+			eventTextContainer,
+			gold
+		} = styles;
+
+		return (
+			<TouchableOpacity
+				style = { eventItem }
+				onPress = { () => { navigation.push('EventDetails', eventRef) } }
```
Comment (idelmr) [inline src/utils/EventPanel.js:50]: Remove unnecessary curly braces

File: src/utils/EventPanel.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+			>
+				<View style = { eventItemInnerContainer }>
+					<View style = { eventTextContainer }>
+						<Text style = { eventTextStyle }>{ committee || type }: { name }</Text>
+						<Text style = { eventTextStyle }>
+							{ convertNumToDate(date) } - { startTime } - { endTime }
```
Comment (idelmr) [inline src/utils/EventPanel.js:56]:
```suggestion
							{ `${convertNumToDate(date)}-${startTime}-${endTime} }
```

**Review by idelmr — CHANGES_REQUESTED** [review]
Looks p-good

File: src/screens/Events/Events.js (hunk 1 above)
Comment (esteban737) [reply src/screens/Events/Events.js:15]: Yea we can probably make that in one up of the upcoming PRs

**Review by idelmr — APPROVED** [review]

**Pull request merged**